### ISSUE TITLE
azurerm_monitor_diagnostic_settings missing categoryGroup messing with resource

### ISSUE THREAD

**Issue by kristinnardal2** [hashicorp/terraform-provider-azurerm#17349]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.3

### AzureRM Provider Version

3.10.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_diagnostic_setting

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.10.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_key_vault" "example" {
  name                = "example-kristinnardal2"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  tenant_id           = "74c901f0-6a97-4cfa-8505-9542ce009b27"
  sku_name            = "standard"
}

resource "azurerm_log_analytics_workspace" "example" {
  name                = "example-workspace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_monitor_diagnostic_setting" "example" {
  name                       = "example"
  target_resource_id         = azurerm_key_vault.example.id
  log_analytics_workspace_id = azurerm_log_analytics_workspace.example.id


  log {
    category = "AuditEvent"
    enabled  = true

    retention_policy {
      enabled = false
    }
  }

  metric {
    category = "AllMetrics"
    enabled  = false

    retention_policy {
      enabled = false
    }
  }
}
```

### Debug Output/Panic Output

```shell
https://gist.github.com/kristinnardal2/41ccf0d845219f32f6cc82af8eec4fc9
```

### Expected Behaviour

I expect Terraform to be able to track changes made to the resource outside of Terraform.

### Actual Behaviour

When changing to a log category group instead of just a log category, Terraform loses the ability to track the resource, and will try to create the object again.

I believe this is because the resource lacks support for category groups, instead it requires that a category is present.
When a category group is chosen, the category is null, which the Terraform module provider can't handle.

I would propose that support for category groups is added, where it's required to use either category groups or categories, not mixing the two.

From the debug logs it also seems that the api version being used does not support the category groups, as it returns a 404 not found for the resource, this would have to be updated.

### Steps to Reproduce

1. Terraform apply
2. In the Azure portal, view the key vault, and click diagnostic settings
3. Change the diagnostic settings to use one of the logs category groups instead of a category
4. Terraform plan

**Comment by teowa** [comment]
@kristinnardal2 , thanks for opening this issue. I have checked this is because `azurerm_monitor_diagnostic_setting` doesn't support for `category groups` yet. This feature is now under development.

**Comment by teowa** [comment]
@kristinnardal2 , this should be supported in PR https://github.com/hashicorp/terraform-provider-azurerm/pull/16367. You can try it in next release (v3.21.0).

**Comment by rcskosir** [comment]
@kristinnardal2 Thanks for opening this issue. Have you been able to try it since release v3.21.0 or higher?

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #16367. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
